### PR TITLE
Fix Weapon reload freeze

### PR DESCRIPTION
**Pull request by Koud-Wind**
## 📝 Description
Fix Unable to find magazine while reload, resulting in weapon freezing

## 🎯 Goals
Weapon reload freeze

## ❌ Non Goals
Code abbreviation is not the goal

## 🚦 Testing 
When reload cannot find the magazine, it will roll back the status

In the creation mode, perform weapon reload, instant sneak, the weapon will not freeze
In survival mode, the weapon reload and instantly discards the only available magazine,  the weapon will not freeze

## ⏮️ Backwards Compatibility 
Fully compatible

## 🖼️ Screenshots/Recordings
https://discord.com/channels/801852948854079489/1078839351468892241/1219634858368696412

## 📖 Added to documentation?

- [ ] 📜 README.md
- [ ] 📑 Documentation
- [ ] 📓 Javadoc
- [ ] 🍕 Comments
- [x] 🙅 No documentation needed

## 😄 [optional] What gif best describes this PR or how it makes you feel?
I discovered this issue in June last year
4}137ZH~VVP6IFUK6REA)95

### CODE REVIEW

**Comment by dark-steveneq** [comment]
In my opinion the reload should proceed "as usual", as in it shouldn't reset your ammo count back to 0 but to full and whatnot

**Comment by Koud-Wind** [comment]
> In my opinion the reload should proceed "as usual", as in it shouldn't reset your ammo count back to 0 but to full and whatnot

It did not reset Ammo to 0, and it should not be reset to full either

**Comment by Desoroxxx** [comment]
Shouldn't get rollbacked either, should just not load this be unloaded

**Comment by Koud-Wind** [comment]
> Shouldn't get rollbacked either, should just not load this be unloaded

I tried to unload it, but the animation between reload and unload didn't connect well
Even in reality, when a soldier's magazine runs out, he should insert it back in

**Comment by Desoroxxx** [comment]
> I tried to unload it, but the animation between reload and unload didn't connect well

You shouldn't animate unload again, when the stall point is reached the magazine is already unloaded, you just need to get it back to ready

**Comment by Koud-Wind** [comment]
> You shouldn't animate unload again, when the stall point is reached the magazine is already unloaded, you just need to get it back to ready

If not rolled back, it will cause the magazine to disappear

**Comment by Desoroxxx** [comment]
Will go back to this after the thing

File: src/main/java/com/paneedah/weaponlib/WeaponReloadAspect.java [hunk 1]
```diff
@@ -402,43 +408,38 @@ public void clientCompoundReload(PlayerWeaponInstance instance) {
...
-                            instance.getWeapon().getRenderer().setMagicMag(instance, nextAttachment, WeaponState.COMPOUND_RELOAD);
-                        else
-							return;
-                        instance.setIsAwaitingCompoundInstructions(true);
-                        stateManager.changeState(this, instance, WeaponState.COMPOUND_REQUESTED, WeaponState.READY);
-                    }
-                }
+                instance.getWeapon().getRenderer().setMagicMag(instance, nextAttachment, WeaponState.COMPOUND_RELOAD);
+                instance.setIsAwaitingCompoundInstructions(true);
+                stateManager.changeState(this, instance, WeaponState.COMPOUND_REQUESTED, WeaponState.READY);
```
[the remark this answers is not in the document]
Comment (Koud-Wind) [reply src/main/java/com/paneedah/weaponlib/WeaponReloadAspect.java:442]: Nothing will change here, increase readability

**Pull request closed without merging**